client/web/src/components/widgets/inputs/SwatchPairInput.vue

```diff
@@ -71,11 +90,47 @@ export default defineComponent({
 			(this.$refs.secondarySwatchPopover as typeof Popover).setOpen();
 			(this.$refs.primarySwatchPopover as typeof Popover).setClosed();
 		},
+		async updatePrimaryColor() {
+			const { update_primary_color, Color } = await wasm;
+
+			let color = this.primaryColor;
+			const button = this.$refs.primaryButton as HTMLButtonElement;
+			button.style.setProperty("--swatch-color", `rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`);
+
+			color = RGB2Floats(this.primaryColor);
+			update_primary_color(new Color(color.r, color.g, color.b, color.a));
+		},
+		async updateSecondaryColor() {
+			const { update_secondary_color, Color } = await wasm;
+
+			let color = this.secondaryColor;
+			const button = this.$refs.secondaryButton as HTMLButtonElement;
+			button.style.setProperty("--swatch-color", `rgba(${color.r}, ${color.g}, ${color.b}, ${color.a})`);
+
+			color = RGB2Floats(this.secondaryColor);
+			update_secondary_color(new Color(color.r, color.g, color.b, color.a));
+		},
 	},
 	data() {
 		return {
 			PopoverDirection,
+			primaryColor: {
```

Comment:
Please reverse these (primary is black, secondary is white) and combine them onto one line each (two lines total).